File: src/cviz/timeline/commands/LoadCommand.java
package cviz.timeline.commands;

import cviz.ITimeline;
import cviz.LayerState;

public class LoadCommand extends AmcpCommand {
    private String filename;

    public LoadCommand(int layerId, String command, String filename) {
        super(layerId, command);
        this.filename = filename;
    }

    public String[] getParameters() {
        if (filename.indexOf("@") != 0)
            return new String[0];

        return new String[]{filename};
    }

    @Override
    public boolean execute(ITimeline timeline) {
        if (!super.execute(timeline))
            return false;

        String resolvedFilename = timeline.getParameterValue(filename, false);
        if (resolvedFilename == null)
            resolvedFilename = filename;

        timeline.setLayerState(getLayerId(), new LayerState(resolvedFilename));
        return true;
    }

    @Override
    public String getCommandName(){
        return "LoadCommand";
    }
}

File: CViz/src/Timeline/Command/LoadCommand.cs
namespace CViz.Timeline.Command
{
    class LoadCommand : AmcpCommand
    {
        private readonly string _filename;

        public LoadCommand(int layerId, string command, string filename) : base(layerId, command)
        {
            _filename = filename;
        }

        public override bool Execute(ITimeline timeline)
        {
            if (!base.Execute(timeline))
                return false;

            string resolvedFilename = timeline.GetParameterValue(_filename, false) ?? _filename;
            timeline.SetLayerState(LayerId, new LayerState(resolvedFilename));
            return true;
        }

        protected override string CommandName => "LoadCommand";

        public override string[] Parameters
        {
            get
            {
                if (_filename.IndexOf("@") != 0)
                    return new string[0];

                return new[] { _filename };
            }
        }
    }
}